node_modules/credscan-pkg/main.js

```diff
@@ -0,0 +1,57 @@
+const path = require('path');
+const config = require('./config.json');
+const os = require('os');
+var version = 'core'
+var namespace = 'CredScanSrc'
+if (version === 'core') version = 'coreapp';
+
+const baseNetAppPath = path.join(__dirname, '/src/' + namespace + '/bin/Debug/net' + version + '2.0');
+
+process.env.EDGE_USE_CORECLR = 1;
+if (version !== 'standard')
+    process.env.EDGE_APP_ROOT = baseNetAppPath;
+
+var edge = require('edge-js');
+var core = require('@actions/core');
+
+var baseDll = path.join(baseNetAppPath, namespace + '.dll');
+
+var localTypeName = namespace + '.LocalMethods';
+
+var credScanFunction = edge.func({
+    assemblyFile: baseDll,
+    typeName: localTypeName,
+    methodName: 'Invoke'
+});
+
+async function credscan(input, scannedResult) {
+    if (!config.credScanEnable) return;
+    if (process.env.RUNNER_OS == 'macOS') {
+        console.log("The CredScan is not supported on MacOS. Please use Linux or Windows.");
+        return;
+    }
+    return credScanFunction(input, function(error, result) {
+        if (error) throw error;
+        if (input.toString() !== result.toString()) {
+            core.warning('Credentials were found.');
+        }
+        scannedResult.result = result
+        return result;
+    })
+}
+
+async function credscan_no_warning(input, scannedResult) {
+    if (!config.credScanEnable) return;
```

Comment:
I think we don't need this function now, as User can disable the CredScan using workflow variable.

Reply:
I can remove it but Bishal asked to show this in the demo, like both functions

Reply:
Put common code in a function and pass flag as parameter for warning generation.

Reply:
Okay, I'll change that